parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -357,43 +373,31 @@ retrycmd_curl_file() {
         fi
     done
 }
-systemctl_restart() {
-    retries=$1; wait_sleep=$2; timeout=$3 svcname=$4
+# base systemctl retry command, should not be called directly - use systemctl_restart, systemctl_stop, systemctl_disable
+_systemctl_retry_svc_operation() {
+    retries=$1; wait_sleep=$2; timeout=$3 operation=$4 svcname=$5 shouldLogRetryInfo=${6:-false}
     for i in $(seq 1 $retries); do
         timeout $timeout systemctl daemon-reload
-        timeout $timeout systemctl restart $svcname && break || \
+        timeout $timeout systemctl $operation $svcname && break || \
         if [ $i -eq $retries ]; then
             return 1
         else
-            systemctl status $svcname --no-pager -l
-            journalctl -u $svcname
+          if [[ "$shouldLogRetryInfo" == "true"  ]]; then
+              systemctl status $svcname --no-pager -l
+              journalctl -u $svcname
+          fi
             sleep $wait_sleep
         fi
     done
 }
+systemctl_restart() {
+    _systemctl_retry_svc_operation "$1" "$2" "$3" "restart" "$4" true
+}
 systemctl_stop() {
-    retries=$1; wait_sleep=$2; timeout=$3 svcname=$4
-    for i in $(seq 1 $retries); do
-        timeout $timeout systemctl daemon-reload
-        timeout $timeout systemctl stop $svcname && break || \
-        if [ $i -eq $retries ]; then
-            return 1
-        else
-            sleep $wait_sleep
-        fi
-    done
+    _systemctl_retry_svc_operation "$1" "$2" "$3" "stop" "$4" 
 }
 systemctl_disable() {
-    retries=$1; wait_sleep=$2; timeout=$3 svcname=$4
-    for i in $(seq 1 $retries); do
-        timeout $timeout systemctl daemon-reload
-        timeout $timeout systemctl disable $svcname && break || \
-        if [ $i -eq $retries ]; then
-            return 1
-        else
-            sleep $wait_sleep
-        fi
-    done
+    _systemctl_retry_svc_operation "$1" "$2" "$3" "disable" "$4" 
 }
 sysctl_reload() {
```

Comment:
can this one be refactored to use the new internal helper as well?

Reply:
yep will make it use the `retrycmd_silent()` helper, didn't realize that it's basically the same.